src/core/net/srp_client.cpp

```diff
@@ -1240,21 +1240,22 @@ Error Client::AppendHostDescriptionInstruction(Message &aMessage, Info &aInfo)
 
     if (mHostInfo.IsAutoAddressEnabled())
     {
-        // Append all addresses on Thread netif excluding link-local and
-        // mesh-local addresses. If no address is appended, we include
+        // Append all preferred addresses on Thread netif excluding link-local
+        // and mesh-local addresses. If no address is appended, we include
```

Comment:
Comment on the last sentence part: clarify this to ML-EID (which is a specific case of mesh-local address that we mean)

```
, we include the Mesh Local EID (ML-EID) address.
```